hazelcast-client/src/test/java/com/hazelcast/client/ClientConnectionTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2008-2013, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.core.Client;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.Collection;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
+public class ClientConnectionTest extends HazelcastTestSupport {
+
+    @After
+    @Before
+    public void cleanup() {
+        HazelcastClient.shutdownAll();
+        Hazelcast.shutdownAll();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testWithIllegalAddress() {
+        String illegalAddress = randomString();
+
+        Hazelcast.newHazelcastInstance();
+        ClientConfig config = new ClientConfig();
+        config.getNetworkConfig().addAddress(illegalAddress);
+        HazelcastClient.newHazelcastClient(config);
+    }
+
+    @Test
+    public void testWithLegalAndIllegalAddressTogether() {
+        String illegalAddress = randomString();
+
+        final HazelcastInstance server = Hazelcast.newHazelcastInstance();
```

Comment:
About finals, they are auto generated in my IDE environment. If we don't have a clear policy about finals, I don't want to change them, because someone else might just say it is better to have them. If we can have a checkstyle rule or something, it can be better to resolve such disagreements,